### PR TITLE
Trans paramters + exports

### PR DESCRIPTION
**Pull request by monzey**
https://github.com/rgsystemes/kb/issues/4613

### CODE REVIEW

File: src/Atoms/Trans/useTranslation.js [hunk 1]
```diff
@@ -5,8 +5,18 @@ export const { Provider } = Context
...
+        translations
+      )
+
+    Object.entries(parameters).forEach(([key, value]) => {
+      translated = translated.replace(new RegExp(`%${key}%`, 'g'), value)
+    })
+
+    return translated
+  }
 }
```
Comment (mesteche) [inline src/Atoms/Trans/useTranslation.js]:
```js
const getTranslationSelector = translations => transKey => (
  transKey
    .split('.')
    .reduce(
      (acc, k) => acc[k] || transKey,
      translations
    )
)

const replaceParam = (translated, [key, value]) => translated.replace(
  new RegExp(`%${key}%`, 'g'),
  value
)

export default () => {
  const translations = useContext(Context) || {}
  const getTranslation = getTranslationSelector(translations)

  return (transKey, parameters = {}) => (
    Object.entries(parameters)
    .reduce(replaceParam, getTranslation(translations))
  )
}
```

Comment (monzey) [reply src/Atoms/Trans/useTranslation.js]: what is the point in doing that ?

Comment (mesteche) [reply src/Atoms/Trans/useTranslation.js]: Splitting the code in smaller, simpler functions.
Although I admit that it's not necessary in this case, it's still a good practice to follow the KISS principle and make it a habit.

Comment (monzey) [reply src/Atoms/Trans/useTranslation.js]: If it's not necessary, I won't do it, feel free to do it in another PR

Comment (mesteche) [reply src/Atoms/Trans/useTranslation.js]: Ok

**Pull request merged**